packages/web-react/src/components/FileUploader/UncontrolledFileUploader.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { useEffect } from 'react';
+import { SpiritUncontrolledFileUploaderProps } from '../../types';
+import { useFileQueue } from './useFileQueue';
+import FileUploader from './FileUploader';
+import FileUploaderInput from './FileUploaderInput';
+import FileUploaderList from './FileUploaderList';
+
+const UncontrolledFileUploader = (props: SpiritUncontrolledFileUploaderProps) => {
+  const {
+    attachmentComponent,
+    maxFileSize,
+    maxUploadedFiles,
+    helperText,
+    inputId,
+    inputLabel,
+    inputName,
+    inputProps,
+    isDisabled,
+    isMultiple,
+    isRequired,
+    labelText,
+    linkText,
+    listId,
+    listProps,
+    onInputError,
+    onChange,
+    errorMessages,
+    validationState,
+    validationText,
+    ...restProps
+  } = props;
+
+  const { fileQueue, addToQueue, clearQueue, onDismiss } = useFileQueue();
+
+  useEffect(() => {
+    if (onChange) {
+      onChange(fileQueue);
+    }
+    // We only want a dependency for fileQueue
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
I think that this disable is not necessary right now when the dependency value is set.

Reply:
Ok, I see, there is also `onChange`. But can we safely add it too? What will change or how does it affect the behavior of the script?

Reply:
Just for info https://www.goldencreche.com/blog/functions-in-dependency-array-of-useeffect. Maybe there can be optimization using a `useCallback` hook. See the article. But nice-to-have.